packages/ui/src/components/form-input/components/form-hoc.tsx

```diff
@@ -1,41 +1,43 @@
-import { ComponentProps, forwardRef } from 'react'
+import { ComponentProps, ComponentType, forwardRef, ForwardRefExoticComponent } from 'react'
 import { Controller, useFormContext } from 'react-hook-form'
 
 type ControllerProps = ComponentProps<typeof Controller>
 type ControllerRenderPropsParamsType = Parameters<NonNullable<ControllerProps['render']>>[0]
+export type WithFormProps = {
+  name: string
+  error?: string
+}
 
 export function withForm<T>(
-  FormPrimitiveComponent: React.ForwardRefExoticComponent<T>,
+  FormPrimitiveComponent: ComponentType<T> | ForwardRefExoticComponent<T>,
```

Comment:
Can we still use `ForwardRefExoticComponent` and avoid `ComponentType<T>`?

Reply:
Nope🫤
ForwardRefExoticComponent doesn't allow to use Select component as a generic, so I faced issues with unresolved `options` type